assets/js/modules/analytics-4/components/common/PropertySelect.js

```diff
@@ -0,0 +1,97 @@
+/**
+ * GA4 Property Select component.
+ *
+ * Site Kit by Google, Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * WordPress dependencies
+ */
+import { useCallback } from '@wordpress/element';
+import { __, sprintf } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import Data from 'googlesitekit-data';
+import { Select, Option } from '../../../../material-components';
+import ProgressBar from '../../../../components/ProgressBar';
+import { STORE_NAME, PROPERTY_CREATE } from '../../datastore/constants';
+import { MODULES_ANALYTICS } from '../../../analytics/datastore/constants';
+import { isValidAccountID } from '../../../analytics/util';
+import { trackEvent } from '../../../../util';
+const { useSelect, useDispatch } = Data;
+
+export default function PropertySelect() {
+	// TODO: Update this select hook to pull accountID from the modules/analytics-4 datastore when GA4 module becomes separated from the Analytics one
+	const accountID = useSelect( ( select ) => select( MODULES_ANALYTICS ).getAccountID() );
+	const properties = useSelect( ( select ) => select( STORE_NAME ).getProperties( accountID ) );
+	const propertyID = useSelect( ( select ) => select( STORE_NAME ).getPropertyID() );
+
+	const isLoading = useSelect( ( select ) => {
+		return ! select( MODULES_ANALYTICS ).hasFinishedResolution( 'getAccounts' ) ||
+			select( STORE_NAME ).isResolving( 'getProperties', [ accountID ] );
```

Comment:
I realize the previous is what is used in the `ProfileSelect` but I think we should probably use `hasFinishedResolution` for consistency, but also this is usually what we want.
```suggestion
			! select( STORE_NAME ).hasFinishedResolution( 'getProperties', [ accountID ] );
```